### PR TITLE
examples: Specify the scope of TLSCertificateDelegation CRD

### LINKED ISSUE #2040: Deploying Contour v1.0.1 using Argo CD becomes a sync error
**What steps did you take and what happened:**

Deploying Contour v1.0.0([these manifests](https://github.com/projectcontour/contour/tree/v1.0.1/examples/contour)) using [Argo CD](https://argoproj.github.io/argo-cd/).

Then `tlscertificatedelegations.contour.heptio.com` CRD became `OutOfSync`.

```console
$ argocd app get contour
Name:               contour
Project:            default
Server:             https://kubernetes.default.svc
Namespace:          projectcontour
URL:                https://<address>/applications/contour
Repo:               https://github.com/<owner>/<repo>.git
Target:             93ba1a00041ef8eec9c03a75f0044cd3ae073a8d
Path:               contour/base
Sync Policy:        Automated (Prune)
Sync Status:        OutOfSync from 93ba1a00041ef8eec9c03a75f0044cd3ae073a8d
Health Status:      Healthy

GROUP                      KIND                      NAMESPACE  NAME                                          STATUS     HEALTH   HOOK  MESSAGE
...
apiextensions.k8s.io       CustomResourceDefinition             httpproxies.projectcontour.io                 Synced     Unknown
apiextensions.k8s.io       CustomResourceDefinition             ingressroutes.contour.heptio.com              Synced     Unknown
apiextensions.k8s.io       CustomResourceDefinition             tlscertificatedelegations.contour.heptio.com  OutOfSync  Unknown
apiextensions.k8s.io       CustomResourceDefinition             tlscertificatedelegations.projectcontour.io   Synced     Unknown
...
```

**What did you expect to happen:**

I expect that Argo CD judges the CRD as `Synced`.

**Anything else you would like to add:**

Because the `spec.scope` of `TLSCertificateDelegation` CRD is empty, Argo CD misjudged the resource status.
So please specify the `Namespaced` in the following field: 

https://github.com/projectcontour/contour/blob/v1.0.1/examples/contour/01-crds.yaml#L360

Argo CD compares the input manifests and the manifests which registered in k8s.
And if there are differences between them, Argo CD judges as `OutOfSync`.

Unfortunately, k8s complements `Namespaced` if CRD's `spec.scope` is blank.
So Argo CD finds the difference like this:

```
$ argocd app diff contour
===== apiextensions.k8s.io/CustomResourceDefinition /tlscertificatedelegations.contour.heptio.com ======
23c23
<   scope: Namespaced
---
>   scope: ""
```

**Environment:**

- Contour version: v1.0.1
- Kubernetes version: (use `kubectl version`): v1.15.3
- Argo CD version: 1.2.5

### PR DESCRIPTION
**Pull request by masa213f**
I explained details in #2040. Please check it.

Fixes #2040

Signed-off-by: masa213f <masa213f@gmail.com>

### CODE REVIEW

File: apis/contour/v1beta1/tlscertificatedelegation.go [hunk 1]
```diff
@@ -42,6 +42,8 @@ type CertificateDelegation struct {
 
 // TLSCertificateDelegation is an TLS Certificate Delegation CRD specificiation.
 // See design/tls-certificate-delegation.md for details.
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=tlscertificatedelegations,singular=tlscertificatedelegation
```
Comment (jpeach) [inline apis/contour/v1beta1/tlscertificatedelegation.go]: Which change ends up generating the "scope" field?

I guess I was expecting you'd have to add the "scope" label to kubebuilder:resource :)

Comment (masa213f) [reply apis/contour/v1beta1/tlscertificatedelegation.go]: Thank you for the review!

`+kubebuilder:resource` marker generates the "scope" field.
If this marker exists, the kubebuilder seems to specify the `scope`'s default value (`Namespaced`).
However, it is good to specify `scope = Namespaced` in this marker. So I will correct it.

"k8s:openapi-gen" was specified in other CRDs, so I added it together. 

By the way, `contour/v1beta1/tlscertificatedelegation.go` and `projectcontour/v1/tlscertificatedelegation.go` still have the following differences, should I unify them?

```
$ diff -u apis/contour/v1beta1/tlscertificatedelegation.go apis/projectcontour/v1/tlscertificatedelegation.go
--- apis/contour/v1beta1/tlscertificatedelegation.go    2019-12-12 23:05:33.496594703 +0000
+++ apis/projectcontour/v1/tlscertificatedelegation.go  2019-12-12 23:03:15.595361314 +0000
@@ -43,7 +43,7 @@
 // TLSCertificateDelegation is an TLS Certificate Delegation CRD specificiation.
 // See design/tls-certificate-delegation.md for details.
 // +k8s:openapi-gen=true
-// +kubebuilder:resource:path=tlscertificatedelegations,scope=Namespaced,singular=tlscertificatedelegation
+// +kubebuilder:resource:path=tlscertificatedelegations,scope=Namespaced,shortName=tlscerts,singular=tlscertificatedelegation
 type TLSCertificateDelegation struct {
        metav1.TypeMeta   `json:",inline"`
        metav1.ObjectMeta `json:"metadata"`

```

Comment (youngnick) [reply apis/contour/v1beta1/tlscertificatedelegation.go]: The CRDs in `contour/v1beta1` are deprecate, please leave them as they are.

Comment (masa213f) [reply apis/contour/v1beta1/tlscertificatedelegation.go]: OK. I just specified the scope by kubebuilder's maker :)

And I changed the `check-stole` rule (in Makefile) a little for checking the auto-generated files of `examples/render` and `examples/contour` at once.

Comment (jpeach) [reply apis/contour/v1beta1/tlscertificatedelegation.go]: What is the Makefile change needed for?

**Comment by masa213f** [comment]
@jpeach 

>Please also add the scope to HTTPProxy in apis/projectcontour/v1/httpproxy.go

OK.

>It's a small issue, but could you please add the scope field at the start so that it doesn't split the resource names?

Thank you. That's important for improving readability!

>What is the Makefile change needed for?

To check both `examples/render` and `examples/contour` at once.

The previous Make rule works with the following steps:

1. Generate `example/render` from `example/contour`
2. Generate `example/contour` by the kubebuilder

Therefore, changes by the kubebuilder are not reflected in `example/ render`.

**Comment by jpeach** [comment]
Merged as 979e1f8

**Pull request closed without merging**